contracts/cw-ics721-bridge/src/ibc_tests.rs

```diff
@@ -0,0 +1,696 @@
+use crate::contract::instantiate;
+use crate::helpers::BATCH_TRANSFER_FROM_CHANNEL_REPLY_ID;
+use crate::helpers::BURN_ESCROW_TOKENS_REPLY_ID;
+use crate::helpers::BURN_SUB_MSG_REPLY_ID;
+use crate::helpers::FAILURE_RESPONSE_FAILURE_REPLY_ID;
+use crate::helpers::INSTANTIATE_AND_MINT_CW721_REPLY_ID;
+use crate::helpers::INSTANTIATE_CW721_REPLY_ID;
+use crate::helpers::INSTANTIATE_ESCROW_REPLY_ID;
+use crate::helpers::MINT_SUB_MSG_REPLY_ID;
+use crate::helpers::TRANSFER_SUB_MSG_REPLY_ID;
+use crate::ibc::ibc_channel_connect;
+use crate::ibc::ibc_channel_open;
+use crate::ibc::ibc_packet_receive;
+use crate::ibc::reply;
+use crate::ibc::NonFungibleTokenPacketData;
+use crate::ibc::IBC_VERSION;
+use crate::msg::ExecuteMsg;
+use crate::msg::InstantiateMsg;
+use crate::ContractError;
+use cosmwasm_std::IbcAcknowledgement;
+use cosmwasm_std::IbcPacketReceiveMsg;
+use cosmwasm_std::{
+    attr,
+    testing::{mock_dependencies, mock_env, mock_info, MockQuerier},
+    to_binary, Binary, ContractResult, DepsMut, Env, IbcChannel, IbcChannelConnectMsg,
+    IbcChannelOpenMsg, IbcEndpoint, IbcOrder, IbcPacket, IbcTimeout, QuerierResult, Reply,
+    Response, SubMsg, SubMsgResponse, SubMsgResult, Timestamp, WasmMsg, WasmQuery,
+};
+
+use crate::ibc_helpers::{ack_fail, ack_success, try_get_ack_error};
+
+// TODO: Check if these are ok, I kinda just stole
+//       them from ICS 20 and ours.
+const CONTRACT_PORT: &str = "wasm.address1";
+const REMOTE_PORT: &str = "stars.address1";
+const CONNECTION_ID: &str = "connection-2";
+const CHANNEL_ID: &str = "channel-1";
+// const DEFAULT_TIMEOUT: u64 = 3600; // 1 hour
+
+const ADDR1: &str = "addr1";
+const CW721_CODE_ID: u64 = 0;
+const ESCROW_CODE_ID: u64 = 1;
+
+fn mock_channel(channel_id: &str) -> IbcChannel {
+    IbcChannel::new(
+        IbcEndpoint {
+            port_id: CONTRACT_PORT.to_string(),
+            channel_id: channel_id.to_string(),
+        },
+        IbcEndpoint {
+            port_id: REMOTE_PORT.to_string(),
+            channel_id: format!("{}5", channel_id),
+        },
+        IbcOrder::Unordered,
+        IBC_VERSION,
+        CONNECTION_ID,
+    )
+}
+
+fn mock_packet(data: Binary) -> IbcPacket {
+    IbcPacket::new(
+        data,
+        IbcEndpoint {
+            port_id: REMOTE_PORT.to_string(),
+            channel_id: CHANNEL_ID.to_string(),
+        },
+        IbcEndpoint {
+            port_id: CONTRACT_PORT.to_string(),
+            channel_id: CHANNEL_ID.to_string(),
+        },
+        42,
```

Comment:
```suggestion
        DEFAULT_TIMEOUT,
```

Reply:
Words are nice. : )

Reply:
Just a suggestion, don't feel strongly about this one. Love the number 42.

Reply:
As a great example of words being nice, it turns out this is actually the packet sequence number lol. Have added a comment explaining what this seemingly random constant is dong.